### PR TITLE
New formula: securefs

### PR DESCRIPTION
**Pull request by netheril96**
A new fuse with transparent authenticated encryption.

### CODE REVIEW

**Comment by DomT4** [comment]
Failing to compile:

```
[ 92%] Building CXX object CMakeFiles/securefs-static.dir/sources/unix.cpp.o
/usr/local/Library/Homebrew/shims/super/clang++   -DHAS_THREAD_LOCAL -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -I/usr/local/include/osxfuse -I/tmp/securefs-20160726-4345-1xojj7t/securefs-0.5.1/sources -I/tmp/securefs-20160726-4345-1xojj7t/securefs-0.5.1/external  -std=c++11 -Wall -Wextra -pedantic -march=native -mtune=native -DNDEBUG   -o CMakeFiles/securefs-static.dir/sources/unix.cpp.o -c /tmp/securefs-20160726-4345-1xojj7t/securefs-0.5.1/sources/unix.cpp
/tmp/securefs-20160726-4345-1xojj7t/securefs-0.5.1/sources/unix.cpp:160:23: error: use of undeclared identifier 'AT_FDCWD'
        if (dir_fd == AT_FDCWD)
                      ^
/tmp/securefs-20160726-4345-1xojj7t/securefs-0.5.1/sources/unix.cpp:168:76: error: use of undeclared identifier 'AT_FDCWD'
FileSystemService::FileSystemService() : impl(new Impl()) { impl->dir_fd = AT_FDCWD; }
                                                                           ^
/tmp/securefs-20160726-4345-1xojj7t/securefs-0.5.1/sources/unix.cpp:172:25: error: use of undeclared identifier 'AT_FDCWD'
    if (impl->dir_fd != AT_FDCWD)
                        ^
/tmp/securefs-20160726-4345-1xojj7t/securefs-0.5.1/sources/unix.cpp:188:16: error: no member named 'openat' in the global namespace; did you mean 'open'?
    int fd = ::openat(impl->dir_fd, path.c_str(), flags, mode);
             ~~^~~~~~
               open
/usr/include/sys/fcntl.h:439:5: note: 'open' declared here
int     open(const char *, int, ...) __DARWIN_ALIAS_C(open);
        ^
/tmp/securefs-20160726-4345-1xojj7t/securefs-0.5.1/sources/unix.cpp:188:23: error: cannot initialize a parameter of type 'const char *' with an lvalue of type 'int'
    int fd = ::openat(impl->dir_fd, path.c_str(), flags, mode);
                      ^~~~~~~~~~~~
/usr/include/sys/fcntl.h:439:22: note: passing argument to parameter here
int     open(const char *, int, ...) __DARWIN_ALIAS_C(open);
                         ^
/tmp/securefs-20160726-4345-1xojj7t/securefs-0.5.1/sources/unix.cpp:197:14: error: no member named 'unlinkat' in the global namespace
    return ::unlinkat(impl->dir_fd, path.c_str(), 0) == 0;
           ~~^
/tmp/securefs-20160726-4345-1xojj7t/securefs-0.5.1/sources/unix.cpp:202:14: error: no member named 'unlinkat' in the global namespace
    return ::unlinkat(impl->dir_fd, path.c_str(), AT_REMOVEDIR) == 0;
           ~~^
/tmp/securefs-20160726-4345-1xojj7t/securefs-0.5.1/sources/unix.cpp:202:51: error: use of undeclared identifier 'AT_REMOVEDIR'
    return ::unlinkat(impl->dir_fd, path.c_str(), AT_REMOVEDIR) == 0;
                                                  ^
/tmp/securefs-20160726-4345-1xojj7t/securefs-0.5.1/sources/unix.cpp:215:16: error: no member named 'mkdirat' in the global namespace; did you mean 'mkdir'?
    int rc = ::mkdirat(impl->dir_fd, path.c_str(), mode);
             ~~^~~~~~~
               mkdir
/usr/include/sys/stat.h:350:5: note: 'mkdir' declared here
int     mkdir(const char *, mode_t);
        ^
/tmp/securefs-20160726-4345-1xojj7t/securefs-0.5.1/sources/unix.cpp:215:52: error: too many arguments to function call, expected 2, have 3
    int rc = ::mkdirat(impl->dir_fd, path.c_str(), mode);
             ~~~~~~~~~                             ^~~~
/usr/include/sys/stat.h:350:1: note: 'mkdir' declared here
int     mkdir(const char *, mode_t);
^
/tmp/securefs-20160726-4345-1xojj7t/securefs-0.5.1/sources/unix.cpp:230:16: error: no member named 'renameat' in the global namespace; did you mean 'rename'?
    int rc = ::renameat(impl->dir_fd, a.c_str(), impl->dir_fd, b.c_str());
             ~~^~~~~~~~
               rename
/usr/include/stdio.h:262:6: note: 'rename' declared here
int      rename (const char *, const char *);
         ^
/tmp/securefs-20160726-4345-1xojj7t/securefs-0.5.1/sources/unix.cpp:230:50: error: too many arguments to function call, expected 2, have 4
    int rc = ::renameat(impl->dir_fd, a.c_str(), impl->dir_fd, b.c_str());
             ~~~~~~~~~~                          ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/stdio.h:262:1: note: 'rename' declared here
int      rename (const char *, const char *);
^
12 errors generated.
make[2]: *** [CMakeFiles/securefs-static.dir/sources/unix.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/securefs-static.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by netheril96** [comment]
Now updated to support Mavericks. Also fixed the errors found by `audit --strict --online`.

File: Formula/securefs.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Securefs < Formula
+  desc "Filesystem with transparent authenticated encryption"
+  homepage "https://github.com/netheril96/securefs"
+  url "https://github.com/netheril96/securefs/archive/0.5.2.tar.gz"
+  sha256 "6feacc286d0b26279366806818d730114966f2106cc4cdb3bde5aef650650fe3"
+  head "https://github.com/netheril96/securefs.git"
+
+  bottle :disable, "The performance may be drastically different when built for the native instead of generic architecture"
```
Comment (DomT4) [inline Formula/securefs.rb]: Don't need this line. We don't add bottles in this tap, and are unlikely to in the future unless existing plans change.

Comment (netheril96) [inline Formula/securefs.rb]: @DomT4 Removed.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-fuse/commit/1057cba65dba2abe3f5c31cdac9bfe7277813e55. Thank you for your contribution to Homebrew @netheril96; we appreciate it! :smiley_cat:

**Pull request merged**